packages/mdc-list/foundation.js

```diff
@@ -198,18 +249,21 @@ class MDCListFoundation extends MDCFoundation {
       if (isRootListItem) {
         if (this.isSingleSelectionList_) {
           // Check if the space key was pressed on the list item or a child element.
-          this.setSelectedIndex(currentIndex);
           this.preventDefaultEvent_(evt);
         }
 
-        // Explicitly activate links, since we're preventing default on Enter, and Space doesn't activate them.
-        this.adapter_.followHref(currentIndex);
-      }
+        const hasCheckboxOrRadio = this.hasCheckboxOrRadioAtIndex_(listItemIndex);
```

Comment:
Can you clarify why these conditions were moved inside the `isRootListItem` check? I assume it has to do with avoiding double-triggers if e.g. a checkbox happens to be focused inside a list item, but I'm less clear on what this does for pure single-selection lists (if it even makes a difference).

Reply:
Checkbox / Radio input's are never reachable using keyboard. The keyboard action (space / enter) happens only on list item root level.

For single selection list case we're not changing anything, it was already inside this condition (`isRootListItem`).